MinecraftClient/Protocol/Handlers/PacketTypes/PacketTypeHandler.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using MinecraftClient.Protocol.Handlers;
+using MinecraftClient.Protocol.Handlers.PacketTypes.Palettes;
+
+namespace MinecraftClient.Protocol.Handlers.PacketTypes
+{
+    public class PacketTypeHandler
+    {
+        private int protocol;
+
+        /// <summary>
+        /// Initialize the handler
+        /// </summary>
+        /// <param name="protocol">Protocol version to use</param>
+        public PacketTypeHandler(int protocol)
+        {
+            this.protocol = protocol;
+        }
+        /// <summary>
+        /// Initialize the handler
+        /// </summary>
+        public PacketTypeHandler() { }
+
+        /// <summary>
+        /// Get the packet type palette
+        /// </summary>
+        /// <returns></returns>
+        public PacketTypePalette GetTypeHandler()
+        {
+            return GetTypeHandler(this.protocol);
+        }
+        /// <summary>
+        /// Get the packet type palette
+        /// </summary>
+        /// <param name="protocol">Protocol version to use</param>
+        /// <returns></returns>
+        public PacketTypePalette GetTypeHandler(int protocol)
```

Comment:
Other palette selections are in `Protocol18.cs`. It's fine to put it here but at least throw an exception as with other palette selections when trying to initialize the palette for a new Minecraft version that is not implemented yet. This helps people trying to implement new Minecraft versions by immediately telling them they forgot the palette 😄

Also, how did you generate the palettes? People implementing newer versions will need guidance for this, so some comments would be useful here to explain how to generate new palettes. (And replace the line `If packet IDs were changed, also update getPacketIncomingType() and getPacketOutgoingID() inside Protocol18PacketTypes.cs` in `Protocol18.cs` to point to the new instructions)